conan/internal/runner/ssh.py

```diff
@@ -1,116 +1,101 @@
+import argparse
 from pathlib import Path
 import pathlib
 import tempfile
 
+from conan.api.conan_api import ConanAPI
 from conan.api.output import Color, ConanOutput
 from conan.errors import ConanException
 
 import os
+import re
 from io import BytesIO
 import sys
 
-def ssh_info(msg, error=False):
-    fg=Color.BRIGHT_MAGENTA
-    if error:
-        fg=Color.BRIGHT_RED
-    ConanOutput().status('\n┌'+'─'*(2+len(msg))+'┐', fg=fg)
-    ConanOutput().status(f'| {msg} |', fg=fg)
-    ConanOutput().status('└'+'─'*(2+len(msg))+'┘\n', fg=fg)
+from conan import conan_version
+from conan.internal.model.version import Version
+from conan.internal.model.profile import Profile
+from conan.internal.runner.output import RunnerOutput
 
 class SSHRunner:
-
-    def __init__(self, conan_api, command, host_profile, build_profile, args, raw_args):
-        from paramiko.config import SSHConfig
-        from paramiko.client import SSHClient
+    def __init__(
+        self,
+        conan_api: ConanAPI,
+        command: str,
+        host_profile: Profile,
+        build_profile: Profile,
+        args: argparse.Namespace,
+        raw_args: list[str],
+    ):
         self.conan_api = conan_api
         self.command = command
         self.host_profile = host_profile
         self.build_profile = build_profile
-        self.remote_host_profile = None
-        self.remote_build_profile = None
-        self.remote_python_command = None
-        self.remote_create_dir = None
-        self.remote_is_windows = None
         self.args = args
         self.raw_args = raw_args
-        self.ssh_config = None
-        self.remote_workspace = None
-        self.remote_conan = None
-        self.remote_conan_home = None
-        if host_profile.runner.get('use_ssh_config', False):
-            ssh_config_file = Path.home() / ".ssh" / "config"
-            ssh_config = SSHConfig.from_file(open(ssh_config_file))
 
-        hostname = host_profile.runner.get("host") # TODO: this one is required
-        if ssh_config and ssh_config.lookup(hostname):
-            hostname = ssh_config.lookup(hostname)['hostname']
+        try:
+            hostname = self._create_ssh_connection()
+        except Exception as e:
+            raise ConanException(f"Error creating SSH connection: {e}")
+        self.logger = ConanOutput()
+        self.logger.set_warnings_as_errors(True)
+        self.runner_logger = RunnerOutput(hostname)
+        self.logger.status(f"Connected to {hostname}", fg=Color.BRIGHT_MAGENTA)
+        self.remote_conn = RemoteConnection(self.client, self.runner_logger)
+
+    def run(self):
+        self._ensure_runner_environment()
+        self._copy_profiles()
+        self._copy_working_conanfile_path()
+        self._remote_create()
 
+    def _create_ssh_connection(self) -> str:
+        from paramiko.config import SSHConfig
```

Comment:
This will need some ``try-except ImportError`` protection with some clarifying messages about the optionality of this.